### ISSUE TITLE
fix inter cluster instance move on file based instances

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#470]
Originally reported of Google Code with ID 414.

```


---- 2013-04-07 05:34:54.452909 start Test tools/move-instance ---------
Checking if instance 'INSTANCE' is not running
IP is unreachable
Command: ../tools/move-instance --verbose --src-ca-file=/tmp/tmpc3663E --src-username=ganeti-qa --src-password-file=/tmp/tmpInvV_i --dest-instance-name=DEST --dest-primary-node=NODE1 --dest-secondary-node=NODE2 --net=0:mac=generate NODE NODE INSTANCE
2013-04-07 05:34:55,092: MainThread INFO Source cluster: NODE
2013-04-07 05:34:55,092: MainThread INFO Destination cluster: NODE
2013-04-07 05:34:55,093: MainThread INFO Instances to be moved: INSTANCE
2013-04-07 05:34:55,093: MainThread INFO Connecting to source RAPI server
2013-04-07 05:34:55,175: MainThread INFO Source cluster RAPI version: 2
2013-04-07 05:34:55,176: MainThread INFO Connecting to destination RAPI server
2013-04-07 05:34:55,260: MainThread INFO Destination cluster RAPI version: 2
2013-04-07 05:34:55,261: Move1 INFO Preparing to move INSTANCE from cluster NODE to NODE as INSTANCE2
2013-04-07 05:34:55,261: Move1 INFO Checking whether instance exists
2013-04-07 05:34:55,385: Move1 INFO Retrieving instance information from source cluster
2013-04-07 05:34:55,827: Move1 INFO Preparing export on source cluster
2013-04-07 05:34:56,167: Move1 INFO [Sun Apr  7 05:34:56 2013] Generating X509 certificate on NODE
2013-04-07 05:34:56,554: Move1 INFO Waiting for destination information to become available
2013-04-07 05:34:56,554: DestForMove1 INFO Creating instance INSTANCE in remote-import mode
2013-04-07 05:34:57,999: DestForMove1 INFO [Sun Apr  7 05:34:57 2013]  - WARNING: Secondary node will be ignored on non-mirrored disk template
2013-04-07 05:34:58,597: DestForMove1 INFO [Sun Apr  7 05:34:58 2013] * disk 0, size 512M
2013-04-07 05:34:58,597: DestForMove1 INFO [Sun Apr  7 05:34:58 2013] * creating instance disks...
2013-04-07 05:34:58,597: DestForMove1 INFO [Sun Apr  7 05:34:58 2013] adding instance INSTANCE to cluster config
2013-04-07 05:34:58,679: DestForMove1 INFO [Sun Apr  7 05:34:58 2013]  - INFO: Waiting for instance INSTANCE to sync disks
2013-04-07 05:34:58,679: DestForMove1 INFO [Sun Apr  7 05:34:58 2013]  - INFO: Instance INSTANCE's disks are in sync
2013-04-07 05:34:58,679: DestForMove1 INFO [Sun Apr  7 05:34:58 2013] * preparing remote import...
2013-04-07 05:35:02,757: DestForMove1 INFO [Sun Apr  7 05:35:02 2013] Disk 0 is now listening
2013-04-07 05:35:02,758: Move1 INFO Starting remote export on source cluster
2013-04-07 05:35:05,250: Move1 ERROR Caught unhandled exception
Traceback (most recent call last):
  File "../tools/move-instance", line 350, in HandleErrors
    fn(*args)
  File "../tools/move-instance", line 586, in __init__
    expinfo["x509_key_name"], mrt.dest_impinfo)
  File "../tools/move-instance", line 658, in _ExportInstance
    (fin_resu, dresults) = poll_job_fn(cl, job_id)[0]
  File "../tools/move-instance", line 410, in PollJob
    remote_import_fn))
  File "/buildbot/qa-kvm/qa-kvm-quick/build/ganeti/rapi/client_utils.py", line 98, in PollJob
    return cli.GenericPollJob(job_id, RapiJobPollCb(rapi_client), reporter)
  File "/buildbot/qa-kvm/qa-kvm-quick/build/ganeti/cli.py", line 1963, in GenericPollJob
    jobs = cbs.QueryJobs([job_id], ["status", "opstatus", "opresult"])
  File "/buildbot/qa-kvm/qa-kvm-quick/build/ganeti/rapi/client_utils.py", line 78, in QueryJobs
    result = self.cl.GetJobStatus(job_ids[0])
  File "/buildbot/qa-kvm/qa-kvm-quick/build/ganeti/rapi/client.py", line 1314, in GetJobStatus
    None, None)
  File "/buildbot/qa-kvm/qa-kvm-quick/build/ganeti/rapi/client.py", line 550, in _SendRequest
    raise GanetiApiError(msg, code=http_code)
GanetiApiError: 500 I do not have access to a valid configuration, cannot process queries: Cannot load configuration: Server got itself in trouble
2013-04-07 05:35:13,050: DestForMove1 INFO Aborting
2013-04-07 05:35:13,050: DestForMove1 WARNING Aborting despite job 813 still running
2013-04-07 05:35:13,051: MainThread INFO Instance move results:
2013-04-07 05:35:13,051: MainThread INFO INSTANCE1 as INSTANCE2: Failed (Source error: 500 I do not have access to a valid configuration, cannot process queries: Cannot load configuration: Server got itself in trouble, destination error: Aborted)
---- 2013-04-07 05:35:13.071704 time=0:00:18.618795 Test tools/move-instance
Traceback (most recent call last):
  File "./ganeti-qa.py", line 786, in <module>
    main()
  File "/buildbot/qa-kvm/qa-kvm-quick/build/ganeti/rapi/client.py", line 245, in wrapper
    return fn(*args, **kwargs)
  File "./ganeti-qa.py", line 781, in main
    RunQa()
  File "./ganeti-qa.py", line 695, in RunQa
    RunInstanceTests()
  File "./ganeti-qa.py", line 616, in RunInstanceTests
    RunExportImportTests(instance, inodes)
  File "./ganeti-qa.py", line 401, in RunExportImportTests
    inodes, tnode)
  File "./ganeti-qa.py", line 91, in RunTest
    retval = fn(*args, **kwargs)
  File "/buildbot/qa-kvm/qa-kvm-quick/build/qa/qa_rapi.py", line 816, in TestInterClusterInstanceMove
    AssertEqual(StartLocalCommand(cmd).wait(), 0)
  File "/buildbot/qa-kvm/qa-kvm-quick/build/qa/qa_utils.py", line 124, in AssertEqual
    raise qa_error.Error("%r == %r" % (first, second))
qa_error.Error: 1 == 0
program finished with exit code 1
elapsedTime=2287.176504

```
Originally added on 2013-04-07 05:50:28 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2013-04-10 15:25:54 +0000 UTC.

Added Labels: Priority-Medium
Added to Milestone: Release2.7

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2013-04-30 14:03:41 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Are there any logs available for this? I mean more than just this output? Thx

Also, any information on how to reproduce this in a non-QA setting? Is it possible to pretend to have 2 rapi cluster when actually only having one cluster?
```
Originally added on 2013-04-30 14:20:26 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Look at what the qa does: it's possible to move from one cluster to itself using the node's hostname, rather than from one cluster to a different one. This way it can be reproduced "easily".
I don't have logs saved, unfortunately, we'd have to dig out which qa run reported it.
Note that if the proper fix is too big, for 2.7 it would be ok for the move tool to just refuse to operate on file based instances rather than crash, and then we can fix it in 2.8+.
Thanks,

Guido
```
Originally added on 2013-04-30 14:54:01 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2013-05-02 13:41:49 +0000 UTC.

Now blocked by: 415 (now #471);

Added to Milestone: Release2.9
Removed from Milestone: Release2.7

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2013-05-02 13:47:19 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Move non-critical bugs scheduled for 2.8 or 2.9 to 2.11, as in those versions only critical bug fixes will be integrated.
```
Originally added on 2013-10-30 09:48:16 +0000 UTC.

Added to Milestone: Release2.11
Removed from Milestone: Release2.9

**Comment by Ganeti-Issues-Migrator** [comment]
```
I am closing this issue because it's old and nobody has updated this in a long time. This is according to our current team effort to migrate to Github issues. Please feel free to comment and re-open this if you think it's still relevant.

See https://groups.google.com/forum/#!topic/ganeti/b2-eukpsXpk for more detail

```
Originally added on 2016-12-12 16:39:19 +0000 UTC.

Changed State: WontFix
Removed from Milestone: Release2.11
